File: src/libs/SspUis.BizLogicLayer/ManualServices/Concrete/ManualService.cs
using GenericServices;
using Microsoft.EntityFrameworkCore;
using SspUis.BizLogicLayer.DualEdu.EnumEduTypeServices;
using SspUis.BizLogicLayer.EnumServices;
using SspUis.BizLogicLayer.Memship;
using SspUis.BizLogicLayer.ModuleServices;
using SspUis.BizLogicLayer.OrganizationServices;
using SspUis.Core.Security;
using SspUis.DataLayer;
using SspUis.DataLayer.EfClasses;
using SspUis.DataLayer.EfClasses.Appeal;
using SspUis.DataLayer.EfClasses.DualEdu;
using SspUis.DataLayer.EfClasses.Hrm;
using SspUis.DataLayer.EfClasses.Proposal;
using StatusGeneric;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using WEBASE;
using WEBASE.EF;
using WEBASE.i18n;
using WEBASE.Models;

namespace SspUis.BizLogicLayer.ManualServices
{
	public class ManualService : StatusGenericHandler, IManualService
	{
		private static bool MODULE_CODE_INITIALIZED = false;

		private readonly ICrudServices _service;
		private readonly DbContext _context;
		private readonly IAuthService _authService;
		private readonly IUnitOfWork _unitOfWork;
		private readonly ICultureHelper _cultureHelper;

		public ManualService(
			ICrudServices crudService,
			DbContext context,
			IAuthService authService,
			IUnitOfWork unitOfWork,
			ICultureHelper cultureHelper)
		{
			_service = crudService;
			_context = context;
			_authService = authService;
			_unitOfWork = unitOfWork;
			_cultureHelper = cultureHelper;
		}

		public IEnumerable<ModuleGroupSelectListDto> GetModuleSelectList()
		{
			if (!MODULE_CODE_INITIALIZED)
			{
				_context.ResolveModules<ModuleCode, Module, ModuleTranslate, ModuleSubGroup, ModuleSubGroupTranslate>();
				MODULE_CODE_INITIALIZED = true;
			}

			return _context.Set<Module>().AsSelectList();
		}

		public SelectList<int> TableSelectList()
		{
			return _context.Set<Table>().AsSelectList();
		}
        public SelectList<long> BankInExcelSelectList()
        {
            return _context.Set<Contractor>().AsBankSelectList();
        }

        public SelectList<int> StateSelectList()
		{
			return _context.Set<State>().AsSelectList();
		}

		public SelectList<int, LanguageSelectListDto> LanguageSelectList()
		{
			return _context.Set<Language>().AsSelectList();
		}
		public SelectList<int> LanguageDegreeSelectList()
		{
			return _context.Set<LanguageDegree>().AsSelectList();
		}
		public SelectList<int> GenderSelectList()
		{
			return _context.Set<Gender>().AsSelectList();
		}
		public SelectList<int> NotificationTypeSelectList()
		{
			return _context.Set<NotificationType>().AsSelectList();
		}

		public SelectList<int> StatusSelectList()
		{
			return _context.Set<Status>().AsSelectList();
		}

		public SelectList<int> SignOrganizationTypeSelectList(SignOrganizationTypeSelectListDtoFilter filter)
		{
			return _context.Set<SignOrganizationType>().AsSelectList(filter);
		}

		public SelectList<int> ApplicantTypeSelectList()
		{
			return _context.Set<ApplicantType>().AsSelectList();
		}
		public SelectList<int> OrganizationGroupSelectList()
		{
			return _context.Set<OrganizationGroup>().AsSelectList();
		}
		public SelectList<int> OrganizationCorruptionGroupSelectList()
		{
			var data = _context.Set<Organization>().Where(a => a.Inn.Equals("201794794") || a.Inn.Equals("305004694") || a.Inn.Equals("207325644")).AsSelectList();
			return data;
		}
		public SelectList<int> BusinessSectorSelectList()
		{
			return _context.Set<BusinessSector>().AsSelectList();
		}

		public SelectList<int> CompanyTypeSelectList()
		{
			return _context.Set<CompanyType>().AsSelectList();
		}

		public SelectList<int> ProposalSubjectSelectList()
		{
			return _context.Set<ProposalSubject>().AsSelectList();
		}

		public SelectList<int> ProposalDisclosureSelectList()
		{
			return _context.Set<ProposalDisclosure>().AsSelectList();
		}

		public SelectList<int> EmploymentTypeSelectList()
		{
			return _context.Set<EmploymentType>().AsSelectList();
		}

		public SelectList<int> RegionSelectList()
		{
			return _context.Set<Region>().AsSelectList();
		}

		public SelectList<int> DistrictSelectList(int? regionId)
		{
			return _context.Set<District>().Where(a => !regionId.HasValue || a.RegionId == regionId)
										   .AsSelectList();
		}

		public SelectList<long> MfySelectList(int? districtId)
		{
			return _context.Set<Mfy>().Where(a => !districtId.HasValue || a.DistrictId == districtId)
									  .AsSelectList();
		}

		public void ClearContractorInfo(int contractorId, bool isOnlyDocument = false)
		{
			try
			{
				_unitOfWork.Context.ClearContractorInfo(contractorId, isOnlyDocument);
			}
			catch (Exception ex)
			{
				AddError($"{ex.Message}: {ex.InnerException}");
			}
		}

		public SelectList<int> GetMonthSelectList()
		{
			TextInfo ti = CultureInfo.CurrentCulture.TextInfo;
			return new SelectList<int>(
				Enumerable.Range(1, 12).Select(month => new SelectListItem<int>
				{
					Value = month,
					Text = ti.ToTitleCase(CultureInfo.GetCultureInfo(_cultureHelper.CurrentCulture.Code).DateTimeFormat.GetMonthName(month))
				}).OrderBy(a => a.Value)
			);
		}

		public SelectList<int> CustomJobTypeSelectList()
		{
			return _context.Set<CustomJobType>().Where(a => a.StateId == StateIdConst.ACTIVE).AsSelectList();
		}

		public SelectList<int> MeetingTypeSelectList()
		{
			return _context.Set<MeetingType>().AsSelectList();
		}

		public SelectList<int> BankCodeSelectList()
		{
			return _context.Set<BankCode>().AsSelectList();
		}

		public SelectList<int> ApplicationModelCodeSelectList()
		{
			return _context.Set<ApplicationModelCode>().AsSelectList();
		}

		public SelectList<int> BirthRegionSelectList()
		{
			return _context.Set<Person>().AsSelectList1();
		}

		public SelectList<int> ApplicationTypeStepSelectList(int applicationTypeId)
		{
			return _context.Set<ApplicationTypeStep>()
				.Include(x => x.Translates)
				.Where(x => x.ApplicationTypeId == applicationTypeId)
				.AsSelectList();
		}

		public SelectList<int> ApplicationTypeSelectList()
		{
			return _context.Set<ApplicationType>()
				.Include(x => x.Translates)
				.AsSelectList();
		}
		public SelectList<int> AppealFormatTypeSelectList()
		{
			return _context.Set<AppealFormatType>().AsSelectList();
		}
		public SelectList<int> AppealTypeSelectList()
		{
			return _context.Set<AppealType>().AsSelectList();
		}
		public SelectList<int> OrganizationAsSelectListByGroup(int[]? groupId)
		{
			groupId ??= new int[0];

			return _context.Set<Organization>()
				.Where(x => groupId.Length == 0 || groupId.Contains(x.OrganizationGroupId))
				.AsSelectList();
		}

		public SelectList<int> ContractorTypeSelectList()
		{
			return _context.Set<ContractorType>().AsSelectList();
		}

		public SelectList<int> RatingSelectList()
		{
			return _context.Set<Rating>().AsSelectList();
		}
		public SelectList<int> OkedTypeSelectList()
		{
			return _context.Set<OkedType>().Include(a => a.Translates).AsSelectList();
		}

		public SelectList<int> AccountNumberSelectList()
		{

			List<string> codes = new List<string>
			{    "20212000003781497001",
				 "20212000803781497025",
				 "20212000503781497051",
				 "20212000803781497026",
				 "20212000903781497007",
				 "20212000803781497022"
			};

			return new SelectList<int>(codes.Select(a => new SelectListItem<int>
			{
				Text = a,
			}));
		}

		public SelectList<int> AsSelectList()
		{
			return _context.Set<EduType>().AsSelectList();
		}
	}
}

File: ssp_uis_frontend/spp_uis_frontend_erp/src/services/others/manual.service.js
import ApiService from '../api.service';

const ManualService = {
   ProposalSubjectSelectList() {
      return ApiService.get('/Manual/ProposalSubjectSelectList');
   },
   ProposalDisclosureSelectList() {
      return ApiService.get('/Manual/ProposalDisclosureSelectList');
   },
   CompanyTypeSelectList() {
      return ApiService.get('/Manual/CompanyTypeSelectList');
   },
   ContractorSelectList() {
      return ApiService.get('/Manual/ContractorSelectList');
   },
   GetModuleSelectList() {
      return ApiService.get('manual/GetModuleSelectList');
   },
   StateSelectList() {
      return ApiService.get('manual/StateSelectList');
   },
   LanguageSelectList() {
      return ApiService.get('manual/LanguageSelectList');
   },
   GenderSelectList() {
      return ApiService.get('manual/GenderSelectList');
   },
   ApplicantTypeSelectList() {
      return ApiService.get('Manual/ApplicantTypeSelectList');
   },
   BusinessSectorSelectList() {
      return ApiService.get('Manual/BusinessSectorSelectList');
   },
   NotificationTypeSelectList() {
      return ApiService.get('manual/NotificationTypeSelectList');
   },
   StatusSelectList() {
      return ApiService.get('manual/StatusSelectList');
   },
   SignOrganizationTypeSelectList(data = {}) {
      return ApiService.post(`manual/SignOrganizationTypeSelectList?incluceBusinessman`, data);
   },
   GetListByDocumentId(tableId, docId) {
      return ApiService.get(`DocumentChangeLog/GetListByDocumentId/${tableId}/${docId}`);
   },
   EmploymentTypeSelectList() {
      return ApiService.get('hrm/manual/EmploymentTypeSelectList');
   },
   WorkScheduleKindSelectList() {
      return ApiService.get('hrm/Manual/WorkScheduleKindSelectList');
   },
   EmpAppointOrderTypeSelectList() {
      return ApiService.get('hrm/manual/EmpAppointOrderTypeSelectList');
   },
   TimesheetIndicatorSelectList() {
      return ApiService.get('hrm/manual/TimesheetIndicatorSelectList');
   },
   TimesheetTypeSelectList() {
      return ApiService.get('hrm/manual/TimesheetTypeSelectList');
   },
   RoundingTypeSelectList() {
      return ApiService.get('hrm/manual/RoundingTypeSelectList');
   },
   MinimumValueTypeSelectList() {
      return ApiService.get('hrm/manual/MinimumValueTypeSelectList');
   },
   CalculateByTimeTypeSelectList() {
      return ApiService.get('hrm/manual/CalculateByTimeTypeSelectList');
   },
   CalculationMethodSelectList() {
      return ApiService.get('hrm/manual/CalculationMethodSelectList');
   },
   CalculationTypeSelectList() {
      return ApiService.get('hrm/manual/CalculationTypeSelectList');
   },
   TariffScaleTypeSelectList() {
      return ApiService.get('hrm/manual/TariffScaleTypeSelectList');
   },
   LimitOperTypeSelectList() {
      return ApiService.get('hrm/manual/LimitOperTypeSelectList');
   },
   ContractorCategorySelectList() {
      return ApiService.get('memship/Manual/ContractorCategorySelectList');
   },
   MemshipContractTypeSelectList() {
      return ApiService.get('memship/Manual/MemshipContractTypeSelectList');
   },
   PositionPeriodSelectList() {
      return ApiService.get('hrm/Manual/PositionPeriodSelectList');
   },
   MediationTypeSelectList() {
      return ApiService.get('claim/Manual/MediationTypeSelectList');
   },
   MediationResultSelectList() {
      return ApiService.get('claim/Manual/MediationResultSelectList');
   },
   ClaimNeedCourtSelectList() {
      return ApiService.get('claim/Manual/ClaimNeedCourtSelectList');
   },
   StaffingTypeSelectList() {
      return ApiService.get('hrm/Manual/StaffingTypeSelectList');
   },
   ClaimApplicationTypeSelectList() {
      return ApiService.get('claim/Manual/ClaimApplicationTypeSelectList');
   },
   CorruptionReviewTypeSelectList() {
      return ApiService.get('corruption/Manual/CorruptionReviewTypeSelectList');
   },
   JoinAntiCorruptionResultTypeSelectList() {
      return ApiService.get('corruption/Manual/JoinAntiCorruptionResultTypeSelectList');
   },
   CustomJobTypeSelectList() {
      return ApiService.get('Manual/CustomJobTypeSelectList');
   },
   BankCodeSelectList() {
      return ApiService.get('Manual/BankCodeSelectList');
   },
   TableSelectList() {
      return ApiService.get('Manual/TableSelectList');
   },
   MeetingTypeSelectList() {
      return ApiService.get('Manual/MeetingTypeSelectList');
   },
   AnswerTypeSelectList() {
      return ApiService.get('quiz/Manual/AnswerTypeSelectList');
   },
   InspectionTypeSelectList() {
      return ApiService.get('quiz/Manual/InspectionTypeSelectList');
   },
   QuestionnaireTypeSelectList() {
      return ApiService.get('quiz/Manual/QuestionnaireTypeSelectList');
   },
   OrganizationGroupSelectList() {
      return ApiService.get('Manual/OrganizationGroupSelectList');
   },
   OrganizationAsSelectListByGroup(id) {
      if (Array.isArray(id)) {
         let url = '/Manual/OrganizationAsSelectListByGroup?';
         id.forEach((e) => {
            url = url + 'groupId=' + e + '&';
         });
         return ApiService.get(url.slice(0, -1));
      } else {
         return ApiService.get(`/Manual/OrganizationAsSelectListByGroup?groupId=${id}`);
      }
   },

   ApplicationModelCodeSelectList() {
      return ApiService.get('Manual/ApplicationModelCodeSelectList');
   },
   BirthRegionSelectList() {
      return ApiService.get('Manual/BirthRegionSelectList');
   },
   GetAgeGroupSelectList() {
      return ApiService.get('Manual/GetAgeGroupSelectList');
   },
   LanguageDegreeSelectList() {
      return ApiService.get('Manual/LanguageDegreeSelectList');
   },

   EmployeeHigherEduDegreeSelectList() {
      return ApiService.get('/hrm/Manual/EmployeeHigherEduDegreeSelectList');
   },
   GetMonthSelectList() {
      return ApiService.get('Manual/GetMonthSelectList');
   },
   ServicePriceTypeSelectList() {
      return ApiService.get('survey/Manual/ServicePriceTypeSelectList');
   },
   ClaimResponsibleTypeSelectList() {
      return ApiService.get('/claim/Manual/ClaimResponsibleTypeSelectList');
   },
   ArbitrationApplicationTypeSelectList() {
      return ApiService.get('Arbitration/Manual/ArbitrationApplicationTypeSelectList');
   },
   ArbitrationCourtSelectList() {
      return ApiService.get('/Arbitration/Manual/ArbitrationCourtSelectList');
   },
   OrderToSendBusinessTripTypeSelectList() {
      return ApiService.get('/hrm/Manual/OrderToSendBusinessTripTypeSelectList');
   },
   TempCalcKindTypeSelectList() {
      return ApiService.get('/hrm/Manual/TempCalcKindTypeSelectList');
   },
   EmployeeSickLeaveTypeSelectList() {
      return ApiService.get('/hrm/Manual/EmployeeSickLeaveTypeSelectList');
   },
   OpfSelectList() {
      return ApiService.get('/memship/Manual/OpfSelectList');
   },
   AppealTypeSelectList() {
      return ApiService.get('/Manual/AppealTypeSelectList');
   },
   AppealFormatTypeSelectList() {
      return ApiService.get('/Manual/AppealFormatTypeSelectList');
   },
   ContractorTypeSelectList() {
      return ApiService.get('/Manual/ContractorTypeSelectList');
   },
   OkedTypeSelectList() {
      return ApiService.get('/Manual/OkedTypeSelectList');
   },
   RatingSelectList() {
      return ApiService.get('/Manual/RatingSelectList');
   },
   EmpAppointOrderTypeSelectList() {
      return ApiService.get('hrm/Manual/EmpAppointOrderTypeSelectList');
   },
   MissedDaysTypeSelectList() {
      return ApiService.get('hrm/Manual/MissedDaysTypeSelectList');
   },
   AccountNumberSelectList() {
      return ApiService.get('Manual/AccountNumberSelectList');
   },
   OrganizationCorruptionGroupSelectList() {
      return ApiService.get('/Manual/OrganizationCorruptionGroupSelectList');
   }
};

export default ManualService;
